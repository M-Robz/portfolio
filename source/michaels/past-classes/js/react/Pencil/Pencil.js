import React from 'react';

/**
 * A pencil banner.
 * @param {Object}  props
 * @param {string}  props.alt - Value for the image's alt attribute.
 * @param {string}  props.srcDesktop - Desktop image source.
 * @param {string}  [props.srcTablet] - Tablet image source.
 * @param {string}  [props.srcMobile] - Mobile image source.
 * @param {string}  [props.href] - The link destination.
 * @param {boolean} [props.outlined] - Whether the pencil is outlined.
 */
export default function Pencil({
    alt, srcDesktop, srcTablet, srcMobile, href, outlined
}) {
    const image = (
        <picture>
            { srcMobile && <source srcSet={srcMobile} media="(max-width: 480px)" /> }
            { srcTablet && <source srcSet={srcTablet} media="(max-width: 800px)" /> }
            <img className="c-pencil__img" src={srcDesktop} alt={alt} />
        </picture>
    );

    return (
        <div className={`c-pencil${ outlined ? ' u-outline' : '' }`}>
            { href ? (
                <a className="c-pencil__link" href={href}>
                    {image}
                </a>
            ) : image }
        </div>
    );
}
