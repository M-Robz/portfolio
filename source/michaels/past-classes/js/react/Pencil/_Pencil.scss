// Dependencies
//   - Mixins: `set-aspect-ratio-container`, `set-aspect-ratio-before`

.c-pencil {
    @include set-aspect-ratio-container(1280, 115);

    @media (max-width: 800px) {
        @include set-aspect-ratio-before(800, 115);
    }

    @media (max-width: 480px) {
        @include set-aspect-ratio-before(480, 115);
    }
}
