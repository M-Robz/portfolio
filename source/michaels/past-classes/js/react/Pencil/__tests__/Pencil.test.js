import React from 'react';
import renderer from 'react-test-renderer';
import Pencil from '../Pencil';

const REQUIRED_PROPS = {
    alt: 'Alt text',
    srcDesktop: 'desktop-image.jpg'
};

const OPTIONAL_PROPS = {
    srcTablet: 'tablet-image.jpg',
    srcMobile: 'mobile-image.jpg',
    href: 'test/link',
    outlined: true
};

it('renders correctly with all props', () => {
    const tree = renderer
        .create(<Pencil {...REQUIRED_PROPS} {...OPTIONAL_PROPS} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with only required props', () => {
    const tree = renderer
        .create(<Pencil {...REQUIRED_PROPS} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
