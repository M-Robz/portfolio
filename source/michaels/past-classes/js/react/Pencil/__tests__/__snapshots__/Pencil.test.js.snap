// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<div
  className="c-pencil u-outline"
>
  <a
    className="c-pencil__link"
    href="test/link"
  >
    <picture>
      <source
        media="(max-width: 480px)"
        srcSet="mobile-image.jpg"
      />
      <source
        media="(max-width: 800px)"
        srcSet="tablet-image.jpg"
      />
      <img
        alt="Alt text"
        className="c-pencil__img"
        src="desktop-image.jpg"
      />
    </picture>
  </a>
</div>
`;

exports[`renders correctly with only required props 1`] = `
<div
  className="c-pencil"
>
  <picture>
    <img
      alt="Alt text"
      className="c-pencil__img"
      src="desktop-image.jpg"
    />
  </picture>
</div>
`;
